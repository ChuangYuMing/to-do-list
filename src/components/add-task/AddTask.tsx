import { Task } from "@/types";
import React, { useState } from "react";

type AddTaskProps = {
  addTask: (task: Task) => void;
};

export default function AddTask({ addTask }: AddTaskProps) {
  const [taskDescription, setTaskDescription] = useState<string>("");

  const handleAddTask = (e: React.MouseEvent) => {
    addTask({
      createdAt: Date.now(),
      description: taskDescription,
      completed: false,
    });

    setTaskDescription("");
  };
  return (
    <div className="add-task-wrap">
      <input
        type="text"
        value={taskDescription}
        onChange={(e) => setTaskDescription(e.target.value)}
        placeholder="Add a new task"
        className="add-task-input"
      />
      <button className="add-task-button" onClick={handleAddTask}>
        Add
      </button>
    </div>
  );
}
