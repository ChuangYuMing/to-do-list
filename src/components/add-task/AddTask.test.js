import React from "react";
import { render } from "@testing-library/react";
import AddTask from "./AddTask";

it("renders correctly", () => {
  const { asFragment } = render(<AddTask />);
  expect(asFragment()).toMatchSnapshot();
});
