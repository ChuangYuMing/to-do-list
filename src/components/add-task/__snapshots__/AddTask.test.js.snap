// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-task-wrap"
  >
    <input
      class="add-task-input"
      placeholder="Add a new task"
      type="text"
      value=""
    />
    <button
      class="add-task-button"
    >
      Add
    </button>
  </div>
</DocumentFragment>
`;
